.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,31 @@
+name: Integration Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  integration_tests:
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Build Packages
+        uses: ./.github/actions/build-packages
+
+      - name: Initialize Host
+        uses: ./.github/actions/host-init
+
+      - name: Build Template
+        uses: ./.github/actions/build-sandbox-template
+
+      - name: Start Services
+        uses: ./.github/actions/start-services
```

Comment:
I really like this, can we maybe put the 4 steps into one composite "prepare cluster"